src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -209,6 +211,7 @@ RazorProjectEngine CreateProjectEngine()
                     builder.SetRootNamespace(HostProject.RootNamespace);
                     builder.SetCSharpLanguageVersion(CSharpLanguageVersion);
                     builder.SetSupportLocalizedComponentNames();
+                    builder.Features.Add(new ConfigureRazorParserOptions(useRoslynTokenizer: ProjectWorkspaceState.UseRoslynTokenizer, CSharpParseOptions.Default));
```

Comment:
The `CSharpParseOptions.Default` is worth calling out here: In the LSP server we do not have access to the `Compilation`, so everything we need from it has to be serialized across the wire from Roslyn. Currently that list is the Tag Helpers, C# lang version, and now the  and we're not currently serializing it across the wire from Roslyn.

@333fred what is the impact of this going to be, and does it matter much?

Serializing the entire CSharpParseOptions is not impossible, but it's a bit annoying keeping up with it. Cohosting has direct access so maybe if the impact is low we don't worry?

Reply:
> @333fred what is the impact of using the default parse options in the IDE going to be, and does it matter much?

Constants defined in the project file won't line up correctly. That's somewhat important.

Reply:
Does that mean we can just send `csharpParseOptions.PreprocessorSymbolNames` across the wire and it will be sufficient? Can we also change the Razor compiler API so it just takes that list rather than the whole parse options object?

Reply:
Alternatively, since they didn't work well previously, can we assume that people aren't desperate for changes around `#if` in Razor files, and the string literal changes are sufficient for dogfooding in 17.13 P1?

Reply:
I think the preprocessor symbols should be sufficient. Not doing for p1 is probably fine, but I do think we want to get them for 17.13 GA.